mlir/lib/Conversion/ArithToEmitC/ArithToEmitC.cpp

```diff
@@ -40,6 +40,29 @@ class ArithConstantOpConversionPattern
   }
 };
 
+/// Check if the signedness of type \p ty matches the expected
+/// signedness, and issue a type with the correct signedness if
+/// necessary.
+Type adaptIntegralTypeSignedness(Type ty, bool needsUnsigned) {
+  if (isa<IntegerType>(ty)) {
+    // Turns signless integers into signed integers.
```

Comment:
I don't understand this comment. The code seems to suggest something like "Treats singless integers as signed integers"?

Reply:
This comment meant that signless integers weren't preserved. We however don't need to say anything about signless integers if we know the spec of the function is to get either a signed or an unsigned type.

Reply:
In the future, we should forbid using signless integers in the emitc dialect, but right now we still need to handle them.

Reply:
They're still allowed, and the new comment is clearer on what happens to them: we get either a signed, or an unsigned equivalent type; shall we mention that the input may be signed, unsigned or signless?

Reply:
For me it's fine, thanks!